File: lib/utils/animations/loop/spin.dart
import '../base/animated_text_state.dart';

AnimatedTextState evaluateSpin(int timeMs, int durationMs) {
  final safeD = durationMs > 0 ? durationMs : 100;
  final progress = (timeMs % safeD).toDouble() / safeD.toDouble();
  return AnimatedTextState(rotation: progress * 360.0);
}

File: android/app/src/main/kotlin/com/typography/animations/loop/Spin.kt
package com.typography.animations.loop

import com.typography.AnimatedTextState

fun evaluateSpin(timeMs: Long, durationMs: Int): AnimatedTextState {
    val progress = (timeMs % durationMs).toFloat() / durationMs.toFloat()
    return AnimatedTextState(rotation = progress * 360f)
}
